### DISCUSSION TITLE
Getting Carla to recognize Hydrogen (with LMMS)

### DISCUSSION THREAD

**Discussion by msea4ever** [hydrogen-music/hydrogen#1280] [Hydrogen on Linux]
I'm a newb and am trying to get Hydrogen to work as a plugin in Carla, which I have installed in LMMS for Lubuntu 20.04. I tried adding all conceivable file paths to Carla but still do not see Hydrogen. I must be missing something. Honestly, I don't even know what TYPE of plugin Hydrogen is, so I just added file paths for each type of plugin.

I do have Hydrogen 1.0.0-rcl working as a standalone program in Lubuntu 20.04.

Any advice on how to get Hydrogen working in Carla with LMMS (Lubuntu 20.04) would be appreciated.

**Comment by theGreatWhiteShark** [comment]
Hey @msea4ever 

For now Hydrogen is only available as standalone application and not as a plugin.

Out of interest: why would you prefer using it as a plugin in Carla over running it next to Carla and LMMS?

**Reply by msea4ever** [reply]
Thanks for your reply. It looks likes someone may have integrated Hydrogen with LMMS to some degree. But I'm not sure. If they used "Jack," this is currently beyond me. 

https://lmms.io/forum/viewtopic.php?t=28822

The main reason I wanted to use Hydrogen in LMMS is because of the former's useful Swing and Humanization dials. I know we can do it manually in LMMS on a 1/192 grid, and this usually sounds better IMO. But sometimes it's just fun to move the dials and hear what you get! 🙂

**Comment by theGreatWhiteShark** [comment]
> If they used "Jack," this is currently beyond me.

Don't be afraid of JACK. There was a time in which doing professional audio under Linux involved patching the system itself. But nowadays things are much more easy and run after installing a couple of packages and setting a couple of parameters in an GUI. See
https://wiki.linuxaudio.org/wiki/introduction or the slightly older https://linuxaudio.github.io/libremusicproduction/html/articles/demystifying-jack-%E2%80%93-beginners-guide-getting-started-jack. 

> The main reason I wanted to use Hydrogen in LMMS is because of the former's useful Swing and Humanization dials. I know we can do it manually in LMMS on a 1/192 grid, and this usually sounds better IMO. But sometimes it's just fun to move the dials and hear what you get!

If you want to use the audio produced by Hydrogen, wiring up Hydrogen and LMMS is the most easiest solution. If you want Hydrogen to send MIDI events and LMMS to produce the audio, things a more complicated.
